seems to fail for me locally:
```
https://github.com/NixOS/nixpkgs/pull/109532

4 packages failed to build:
python37Packages.tokenizers python37Packages.transformers python38Packages.tokenizers python38Packages.transformers
```

```
/nix/store/yi9a1fafijn0nf5m8qkmnl955vi2mncm-python3-3.8.6/lib/python3.8/os.py:213: in makedirs
    makedirs(head, exist_ok=exist_ok)
/nix/store/yi9a1fafijn0nf5m8qkmnl955vi2mncm-python3-3.8.6/lib/python3.8/os.py:223: in makedirs
    mkdir(name, mode)
E   PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
```

--- a/pkgs/development/python-modules/tokenizers/default.nix
+++ b/pkgs/development/python-modules/tokenizers/default.nix
@@ -7,6 +7,7 @@
 , wheel
 , numpy
 , python
+, datasets
 , pytestCheckHook
 , requests
 }:
@@ -50,16 +51,16 @@ let
   };
 in rustPlatform.buildRustPackage rec {
   pname = "tokenizers";
-  version = "0.9.4";
+  version = "0.10.0";
 
   src = fetchFromGitHub {
     owner = "huggingface";
     repo = pname;
     rev = "python-v${version}";
-    hash = "sha256-JXoH9yfhMIFg5qDY5zrF6iWb7XKugjMfk1NxSizfaWg=";
+    hash = "sha256-rQ2hRV52naEf6PvRsWVCTN7B1oXAQGmnpJw4iIdhamw=";
   };
 
-  cargoSha256 = "sha256-u9qitrOxJSABs0VjwHUZgmw7VTQXNbp6l8fKKE/RQ7M=";
+  cargoSha256 = "sha256-BoHIN/519Top1NUBjpB/oEMqi86Omt3zTQcXFWqrek0=";
 
   sourceRoot = "source/bindings/python";
 
@@ -75,6 +76,7 @@ in rustPlatform.buildRustPackage rec {
   ];
 
   installCheckInputs = [
+    datasets
     pytestCheckHook
     requests
   ];
@@ -104,6 +106,15 @@ in rustPlatform.buildRustPackage rec {
   installPhase = ''
     pipInstallPhase
   '';
+
+  preCheck = ''
+    HOME=$TMPDIR
+  '';
+
+  disabledTests = [
+    # Downloads data using the datasets module.
+    "TestTrainFromIterators"
+  ];
 
   meta = with lib; {
     homepage = "https://github.com/huggingface/tokenizers";

